src/PowerShellEditorServices/Services/Symbols/ReferenceTable.cs

```diff
@@ -55,64 +71,146 @@ internal void EnsureInitialized()
         }
 
         public override AstVisitAction VisitVariableExpression(VariableExpressionAst variableExpressionAst)
         {
             // TODO: Consider tracking unscoped variable references only when they declared within
             // the same function definition.
             _references.AddReference(
+                SymbolType.Variable,
                 $"${variableExpressionAst.VariablePath.UserPath}",
-                variableExpressionAst.Extent);
+                variableExpressionAst.Extent
+            );
+
+            return AstVisitAction.Continue;
+        }
+
+        public override AstVisitAction VisitTypeDefinition(TypeDefinitionAst typeDefinitionAst)
+        {
+            SymbolType symbolType = typeDefinitionAst.IsEnum ? SymbolType.Enum : SymbolType.Class;
+
+            // We only want the type name. Get start-location for name
+            IScriptExtent nameExtent = VisitorUtils.GetNameExtent(typeDefinitionAst);
+            _references.AddReference(symbolType, typeDefinitionAst.Name, nameExtent);
+
+            return AstVisitAction.Continue;
+        }
+
+        public override AstVisitAction VisitTypeExpression(TypeExpressionAst typeExpressionAst)
+        {
+            _references.AddReference(
+                SymbolType.Type,
+                typeExpressionAst.TypeName.Name,
+                typeExpressionAst.Extent);
+
+            return AstVisitAction.Continue;
+        }
+
+        public override AstVisitAction VisitTypeConstraint(TypeConstraintAst typeConstraintAst)
+        {
+            _references.AddReference(SymbolType.Type, typeConstraintAst.TypeName.Name, typeConstraintAst.Extent);
+
+            return AstVisitAction.Continue;
+        }
+
+        public override AstVisitAction VisitFunctionMember(FunctionMemberAst functionMemberAst)
+        {
+            SymbolType symbolType = functionMemberAst.IsConstructor
+                ? SymbolType.Constructor
+                : SymbolType.Method;
+
+            // We only want the method/ctor name. Get start-location for name
+            IScriptExtent nameExtent = VisitorUtils.GetNameExtent(functionMemberAst, true, false);
+            _references.AddReference(
+                symbolType,
+                VisitorUtils.GetMemberOverloadName(functionMemberAst, true, false),
+                nameExtent);
 
             return AstVisitAction.Continue;
         }
+
+        public override AstVisitAction VisitPropertyMember(PropertyMemberAst propertyMemberAst)
+        {
+            // We only want the property name. Get start-location for name
+            IScriptExtent nameExtent = VisitorUtils.GetNameExtent(propertyMemberAst, false);
+            _references.AddReference(
+                SymbolType.Property,
+                VisitorUtils.GetMemberOverloadName(propertyMemberAst, false),
+                nameExtent);
+
+            return AstVisitAction.Continue;
+        }
+
+        // TODO: What else can we implement?
```

Comment:
> Could we potentially identify when the "reference" is a definition, save that to the `SymbolReference`, and then use it later instead of searching all over again?

I suggested a `IsDefinition`-prop at one point here also. https://github.com/PowerShell/PowerShellEditorServices/issues/1889#issuecomment-1215560718

Reply:
This came along nicely!